bundles/org.openhab.binding.modbus.studer/src/main/java/org/openhab/binding/modbus/studer/internal/StuderParser.java

```diff
@@ -0,0 +1,233 @@
+/**
+ * Copyright (c) 2010-2020 Contributors to the openHAB project
+ *
+ * See the NOTICE file(s) distributed with this work for additional
+ * information.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ */
+package org.openhab.binding.modbus.studer.internal;
+
+import org.eclipse.jdt.annotation.NonNullByDefault;
+import org.eclipse.jdt.annotation.Nullable;
+
+/**
+ * The {@link StuderParser} class with helper method
+ * and possible values for mode and state
+ *
+ * @author Giovanni Mirulla - Initial contribution
+ */
+@NonNullByDefault
+public class StuderParser {
+    public enum ModeXtender {
+        INVALID(0),
+        INVERTER(1),
+        CHARGER(2),
+        BOOST(3),
+        INJECTION(4),
+        UNKNOWN(-1);
+
+        private final int code;
+
+        ModeXtender(int code) {
+            this.code = code;
+        }
+
+        public int code() {
+            return this.code;
+        }
+    }
+
+    public static ModeXtender getModeXtenderByCode(int code) {
+        switch (code) {
+            case 0:
+                return ModeXtender.INVALID;
+            case 1:
+                return ModeXtender.INVERTER;
+            case 2:
+                return ModeXtender.CHARGER;
+            case 3:
+                return ModeXtender.BOOST;
+            case 4:
+                return ModeXtender.INJECTION;
+            default:
+                return ModeXtender.UNKNOWN;
```

Comment:
It looks that in cases the value is unknown the channel should be set to `UnDefType.UNDEF`. That is the generic indicator the status is unknown. You seem to create a custom undefined state here. Same comment for the other enums. I also see this because the unknown state isn't mentioned in the options list in the thing xml files. That seem to suggest the unknown state should be the undef state I mentioned.

Reply:
I created it because the studer devices have `UNKNOWN` state and because when I pass the Mode on `updateState`, I pass as StringType. I change it and now when the `UNKNOWN` state occurs the `updateState` has `UnDefType.UNDEF`